tests/backup/backup_psa_test.go

```diff
@@ -943,3 +944,440 @@ var _ = Describe("{RestoreFromHigherPrivilegedNamespaceToLower}", Label(TestCase
+			log.FailOnError(err, "Switching context to source cluster failed")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			log.Infof("The restricted backup which is going to restore is %v", restrictedManualBackupName)
+			restoreName := fmt.Sprintf("restore-rtob-%v", RandomString(5))
+			err = CreateRestoreWithProjectMapping(restoreName, restrictedManualBackupName, namespaceMapRestrictedToBaseline, DestinationClusterName, BackupOrgID, ctx, nil, restoreProjectUIDMapping, restoreProjectNameMapping)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of restore with %s from backup %s", restoreName, restrictedManualBackupName))
+		})
+
+		Step("Restore baseline to privilege namespaces with namespace & project mappings", func() {
+			log.InfoD("Restore baseline to privilege namespaces with namespace & project mappings")
+			log.Infof("The baseline backup which is going to restore is %v", baselineManualBackupName)
+			restoreName := fmt.Sprintf("restore-btop-%v", RandomString(5))
+			err = CreateRestoreWithProjectMapping(restoreName, baselineManualBackupName, namespaceMapBaselineToPrivilege, DestinationClusterName, BackupOrgID, ctx, nil, restoreProjectUIDMapping, restoreProjectNameMapping)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of restore with %s from backup %s", restoreName, baselineManualBackupName))
+		})
+
+		Step("Creating schedule backup with all restricted & baseline namespaces", func() {
+			log.InfoD("Creating schedule backup with all restricted & baseline namespaces")
+			schPolicyUid, _ = Inst().Backup.GetSchedulePolicyUid(BackupOrgID, ctx, periodicPolicyName)
+			schBackupName := fmt.Sprintf("schbkp-all-ns-%v", RandomString(4))
+			scheduleNames = append(scheduleNames, schBackupName)
+			labelSelectors := make(map[string]string)
+			log.InfoD("Creating a schedule backup [%s] with namespaces [%s]", backupName, bkpNamespaces)
+			allNSScheduleBackup, err = CreateScheduleBackupWithValidation(ctx, schBackupName, SourceClusterName, backupLocationName, backupLocationUID, scheduledAppContexts, labelSelectors, BackupOrgID, preRuleName, preRuleUid, postRuleName, postRuleUid, periodicPolicyName, schPolicyUid)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of schedule backup [%s] with all namespaces contains both restricted and baseline apps.", allNSScheduleBackup))
+			schBackupList = append(schBackupList, allNSScheduleBackup)
+			log.InfoD("Schedule backup list after all namespaces backup %v", schBackupList)
+		})
+
+		wg.Add(1)
+		go func() {
+			defer GinkgoRecover()
+			defer wg.Done()
+			Step("Creating schedule backup with restricted namespaces", func() {
+				log.InfoD("Creating schedule backup with restricted namespaces")
+				schPolicyUid, _ = Inst().Backup.GetSchedulePolicyUid(BackupOrgID, ctx, periodicPolicyName)
+				schBackupName := fmt.Sprintf("schbkp-restricted-%v", RandomString(5))
+				mutex.Lock()
+				scheduleNames = append(scheduleNames, schBackupName)
```

Comment:
Can we do it in a single mutex lock? Yes it will save only a few milliseconds but would be a better way to do.